vertx-lang-js-gen/src/test/resources/vertx-js/future.js

```diff
@@ -0,0 +1,414 @@
+/*
+ * Copyright 2014 Red Hat, Inc.
+ *
+ * Red Hat licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+/** @module vertx-js/future */
+var utils = require('vertx-js/util/utils');
+var Promise = require('vertx-js/promise');
+
+var io = Packages.io;
+var JsonObject = io.vertx.core.json.JsonObject;
+var JsonArray = io.vertx.core.json.JsonArray;
+var JFuture = Java.type('io.vertx.core.Future');
+
+/**
```

Comment:
is it generated or is it copied from the generated code ?

Reply:
copied from generated core. It's needed for test purposes

Reply:
excellent, can you add a note about it ? like "copied from this generated file" and give path of file to copy